src/coreclr/vm/ilmarshalers.cpp

```diff
@@ -1959,7 +1959,7 @@ void ILCUTF8Marshaler::EmitConvertContentsCLRToNative(ILCodeStream* pslILEmit)
     STANDARD_VM_CONTRACT;
 
     if (m_dwInstance == LOCAL_NUM_UNUSED)
-        m_dwInstance = pslILEmit->NewLocal(LocalDesc(CoreLibBinder::GetClass(CLASS__UTF8STRINGMARSHALLER)));
+        m_dwInstance = pslILEmit->NewLocal(LocalDesc(CoreLibBinder::GetClass(CLASS__UTF8STRINGMARSHALLER_IN)));
```

Comment:
We need to initialize this only when `bPassByValueInOnly` is true now. The remaining cases are stateless and do not need the instance.

Reply:
(same in the AOT compiler version)